django/forms/models.py

```diff
@@ -1211,6 +1211,7 @@ def _get_foreign_key(parent_model, model, fk_name=None, can_fail=False):
         if len(fks_to_parent) == 1:
             fk = fks_to_parent[0]
             parent_list = parent_model._meta.get_parent_list()
+            parent_list.append(parent_model)
```

Comment:
I think we should add the same to the second branch when `fk_name` is not defined:
```diff
diff --git a/django/forms/models.py b/django/forms/models.py
index b80e188a23..6a2608c0b3 100644
--- a/django/forms/models.py
+++ b/django/forms/models.py
@@ -1237,6 +1237,7 @@ def _get_foreign_key(parent_model, model, fk_name=None, can_fail=False):
     else:
         # Try to discover what the ForeignKey from model to parent_model is
         parent_list = parent_model._meta.get_parent_list()
+        parent_list.append(parent_model)
         fks_to_parent = [
             f
             for f in opts.fields
```
Please also add a regression test for this case.

Reply:
Yep you're right! Added the test and update on the second commit